src/System.Private.CoreLib/shared/System/Runtime/InteropServices/GCHandle.cs

```diff
@@ -0,0 +1,164 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Internal.Runtime.CompilerServices;
+#if BIT64
+using nint = System.Int64;
+#else
+using nint = System.Int32;
+#endif
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Represents an opaque, GC handle to a managed object. A GC handle is used when an
+    /// object reference must be reachable from unmanaged memory.
+    /// </summary>
+    /// <remarks>
+    /// There are 4 kinds of roots:
+    /// Normal: Keeps the object from being collected.
+    /// Weak: Allows object to be collected and handle contents will be zeroed.
+    /// Weak references are zeroed before the finalizer runs, so if the
+    /// object is resurrected in the finalizer the weak reference is still zeroed.
+    /// WeakTrackResurrection: Same as Weak, but stays until after object is really gone.
+    /// Pinned - same as Normal, but allows the address of the actual object to be taken.
+    /// </remarks>
+    [StructLayout(LayoutKind.Sequential)]
+    public partial struct GCHandle
+    {
+        // IMPORTANT: This must be kept in sync with the GCHandleType enum.
+        private const GCHandleType MaxHandleType = GCHandleType.Pinned;
```

Comment:
I think inlining the enum GCHandleType.Pinned would be better

Reply:
What do you mean by that?

Reply:
It's used at a single location to check the enum range, using GCHandleType.Pinned there instead could be better and I think that's how enum values are checks elsewhere too (but didn't check that)

Reply:
Ok